searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_iterator.cpp

```diff
@@ -29,7 +30,7 @@ is_compatible(const vespalib::eval::ValueType& lhs,
  * Keeps a heap of the K best hit distances.
  * Currently always does brute-force scanning, which is very expensive.
  **/
-template <bool strict>
+template <bool strict, bool no_bitvector>
```

Comment:
Consider inverting template argument to _use_filter_ instead. It is easier to read it that way.